src/indexer/services/bitcoinIndexer/bitcoinIndexer.ts

```diff
@@ -0,0 +1,113 @@
+/*
+The Licensed Work is (c) 2023 Sygma
+SPDX-License-Identifier: LGPL-3.0-only
+*/
+
+import winston from "winston"
+import { Height, RPCClient } from "rpc-bitcoin"
+import { logger as rootLogger } from "../../../utils/logger"
+import { sleep } from "../../../indexer/utils/substrate"
+import { Domain } from "../../config"
+import DomainRepository from "../../repository/domain"
+import { saveEvents } from "../../../indexer/utils/bitcoin"
+import ExecutionRepository from "../../../indexer/repository/execution"
+import DepositRepository from "../../../indexer/repository/deposit"
+import TransferRepository from "../../../indexer/repository/transfer"
+import FeeRepository from "../../../indexer/repository/fee"
+import CoinMarketCapService from "../coinmarketcap/coinmarketcap.service"
+import { Block } from "./bitcoinTypes"
+
+const BLOCK_TIME = Number(process.env.BLOCK_TIME) || 12000
+const BLOCK_DELAY = Number(process.env.BLOCK_DELAY) || 3
+
+export class BitcoinIndexer {
+  private domainRepository: DomainRepository
+  private executionRepository: ExecutionRepository
+  private depositRepository: DepositRepository
+  private transferRepository: TransferRepository
+  private feeRepository: FeeRepository
+  private domain: Domain
+  private logger: winston.Logger
+  private stopped = false
+  private client: RPCClient
+  private coinMarketCapService: CoinMarketCapService
+
+  constructor(
+    domainRepository: DomainRepository,
+    domain: Domain,
+    executionRepository: ExecutionRepository,
+    depositRepository: DepositRepository,
+    transferRepository: TransferRepository,
+    feeRepository: FeeRepository,
+    coinMarketCapService: CoinMarketCapService,
+  ) {
+    this.domainRepository = domainRepository
+    this.executionRepository = executionRepository
+    this.transferRepository = transferRepository
+    this.depositRepository = depositRepository
+    this.feeRepository = feeRepository
+    this.domain = domain
+    this.coinMarketCapService = coinMarketCapService
+
+    const url = process.env.BTC_URL
```

Comment:
Let's try to avoid having env inside objects. 
Keep the env in inside the init fuction (inside `indexer/index.ts`) and pass a config object.